airbyte-lib/pyproject.toml

```diff
@@ -45,7 +45,7 @@ ruff = "^0.1.11"
 types-jsonschema = "^4.20.0.0"
 google-cloud-secret-manager = "^2.17.0"
 types-requests = "2.31.0.4"
-freezegun = "^1.4.0"
+source-faker = {path = "../airbyte-integrations/connectors/source-faker", develop = true}
```

Comment:
Before deciding to include, I checked the dependencies of source-faker. It only depends on memesis and the CDK. And memesis has [no hard dependencies](https://mimesis.name/en/master/installation.html#dependencies).

Given the light footprint, and that this is only used as a test dependency, the benefits are:

1. Stateless execution. There's no time at which the connector is not installed.
2. Faster test invocation. Tests can start instantly without having to install of verify the source's installation fixture is ready.
3. Simplicity. Following from the statelessness of not needing the fixtures to install it, the overall test framework is simpler. We can expect that `source-faker` is on `PATH` and if not, its an error.
4. Tests BYO model for connector executables.

I'm glad we have the `source-faker` example which installs itself and valides that part of the process. Given that we already have those checks, adding as a dev dependency just focuses on the value we want to get from this connector (running it) without additional complexity.

Sidebar: I see that `source-faker` is now [published to pypi](https://pypi.org/project/airbyte-source-faker/) 🙌, so we can pin to that specific published pypi version instead of using the relative reference. This will ensure our tests are stable even if the faker source changes.